bananalyzer/runner/runner.py

```diff
@@ -22,25 +20,42 @@ class BananalyzerTest(BaseModel):
 class TestGenerator:
     def __init__(self) -> None:
         self._names: Dict[str, int] = {}
+        self._classnames: Dict[str, int] = {}
 
-    def generate_test(self, example: Example, headless: bool) -> BananalyzerTest:
+    def generate_test(self, example: Example) -> BananalyzerTest:
         return BananalyzerTest(
             code=f"""
 @pytest.mark.asyncio
-async def {self._generate_name(example)}() -> None:
-    async with async_playwright() as p:
-        browser = await p.chromium.launch(headless={headless})
-        context = await browser.new_context()
-        example = get_example_by_url("{example.url}")
+class {self._generate_class_name(example)}:
+
+    @classmethod
+    def setup_class(cls):
+        cls.example = get_example_by_url("{example.url}")
+
+
+    @pytest_asyncio.fixture(scope="class")
+    async def result(self, context, agent):
+        yield await agent.run(context, self.example)
 
-        # The agent is imported into the global context prior to this call
-        result = await agent.run(context, example)
-        for curr_eval in example.evals:
-            curr_eval.eval_results(p, result)
-    """,
+    {"".join(self._generate_eval_test(eval_, i) for i, eval_ in enumerate(example.evals))}
+""",
             example=example,
         )
 
+    def _generate_eval_test(self, eval_: Eval, i: int) -> str:
+        if eval_.type == "json_match" and isinstance(eval_.expected, dict):
+            return f"""
+    @pytest.mark.parametrize("key", {list(eval_.expected.keys())})
+    async def test_match_field(self, key, result) -> None:
+        assert self.example.evals[{i}].expected[key] == result[key]
+
```

Comment:
nit: Ideally we can encapsulate this stuff with the eval somewhere. Will need to change two places now whenever evals are added/deleted

Reply:
so you think evals should be responsible for code gen too?